### ISSUE TITLE
🐛 Bug Report: Accessibility issues with FavoriteEntity label

### ISSUE THREAD

**Issue by kmikko** [backstage/backstage#26033]
### 📜 Description

`FavoriteEntity`'s title is always set to `favorite` which I think should rather reflect state, i.e. if entity is favorited then title should say `Remove from favorites` or when it's not favorited when title should say `Add to favorites`. There are also problems selecting said component using role based selectors because title is not rendered when it's being hovered. This has caused some issues and flakiness with end-to-end tests that we've written using Playwright. Issue is also present using screen reader; there's no feedback when entity is favorited/un-favorited.

### 👍 Expected behavior

Appropriate label, either `Add to favorites` or `Remove from favorites`, should be present in all times and reflect this when used with screen reader.

### 👎 Actual Behavior with Screenshots

`FavoriteEntity` cannot be selected using role based selectors when it's active or being hovered:
<img width="527" alt="Screenshot 2024-08-15 at 11 16 08" src="https://github.com/user-attachments/assets/841a0ab3-98a4-40fb-bbbf-2105f274bd04">

<img width="844" alt="Screenshot 2024-08-15 at 13 55 57" src="https://github.com/user-attachments/assets/d0089791-4ea8-4609-9cde-a8c4569376dd">

No screen reader feedback when added to / removed from favorites:
<img width="1101" alt="Screenshot 2024-08-15 at 14 04 16" src="https://github.com/user-attachments/assets/e9e6e131-ee69-4cc9-90a0-030260f060a2">

<img width="1098" alt="Screenshot 2024-08-15 at 14 04 26" src="https://github.com/user-attachments/assets/cb605db5-85ae-4d4c-8927-2382447b0791">

### 👟 Reproduction steps

Go to: https://demo.backstage.io/catalog/default/component/backstage
Use browser's accessibility mode and/or toggle screen reader and toggle entity's favorite status by clicking the star icon.

### 📃 Provide the context for the Bug.

I think this could be fixed by providing [id](https://github.com/mui/material-ui/blob/v4.12.2/packages/material-ui/src/Tooltip/Tooltip.js#L577-L581) to MUI Tooltip as otherwise random generated ID [will be used](https://github.com/mui/material-ui/blob/v4.12.2/packages/material-ui/src/Tooltip/Tooltip.js#L418-L425) which breaks aria label.

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!
